lib/idn.c

```diff
@@ -50,6 +50,69 @@
 #include "curl_memory.h"
 #include "memdebug.h"
 
+#if defined(__APPLE__)
+/* for macOS and iOS targets */
+#include <unicode/uidna.h>
+
+static CURLcode mac_idn_to_ascii(const char *in, char **out)
+{
+  UErrorCode err = U_ZERO_ERROR;
+  UIDNA* idna = uidna_openUTS46(UIDNA_CHECK_BIDI, &err);
+  if(U_FAILURE(err)) {
+    return CURLE_OUT_OF_MEMORY;
+  }
+  else {
+    UIDNAInfo info = UIDNA_INFO_INITIALIZER;
+    char buffer[256] = {0};
+    int32_t inlength = -1; /* zero terminated */
+
+    int32_t len = uidna_nameToASCII_UTF8(idna, in, inlength, buffer,
+      sizeof(buffer), &info, &err);
+    uidna_close(idna);
+    if(U_FAILURE(err)) {
+      return CURLE_URL_MALFORMAT;
+    }
+    else {
+      *out = strdup(buffer);
+      fprintf(stderr, "%s: %s\n", __FUNCTION__, *out);
```

Comment:
Suggesting to replace `__FUNCTION__` with a literal. Same in the next hunk.

Reply:
fprintf() is not allowed though. infof() or failf() should be used for logging/errors.

Reply:
Of course. Removed. I just had them to print the input/output for debugging.
And I thought I removed them all...